src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -524,6 +524,80 @@ public static unsafe SocketError ReceiveMessageFrom(Socket socket, SafeSocketHan
             return SocketError.Success;
         }
 
+        public static unsafe SocketError ReceiveMessageFrom(Socket socket, SafeSocketHandle handle, Span<byte> buffer, ref SocketFlags socketFlags, Internals.SocketAddress socketAddress, out Internals.SocketAddress receiveAddress, out IPPacketInformation ipPacketInformation, out int bytesTransferred)
```

Comment:
Code duplication seems unnecessary in Windows PAL, since we do the pinning the same way in both methods. We only need a single span variant.